solidity/contracts/factory.sol

```diff
@@ -20,54 +20,67 @@ import {IZetoFungibleInitializable} from "./lib/interfaces/zeto_fungible_initial
 import {IZetoNonFungibleInitializable} from "./lib/interfaces/zeto_nf_initializable.sol";
 
 contract ZetoTokenFactory {
+    struct InitArgs {
+        address implementation;
+        address depositVerifier;
+        address withdrawVerifier;
+        address verifier;
```

Comment:
These are optional constructor parameters and if feels like you need to guess if you are either going to call `deployZetoNonFungibleToken` only pass the verifier and `deployZetoFungibleToken` you need to pass all three. The previous code imposed that on you at the function declaration. So maybe `registerFungibleImplementation` and `registerNonFungibleImplementation` not sure..

or are you saying they could call either for the same implementation?

Reply:
thanks for pointing this out. so the deposit and withdraw verifiers are optional for non-fungible tokens. I settled on still using the same object and only requiring the `implementation` and `verifier` on registration, because both are required, but checking the `depositVerifier` and `withdrawVerifier` inside `deployZetoFungibleToken()`.

When registering for a non-fungible token, the optional addresses can be set to "0x0000000000000000000000000000000000000000", as shown in test/factory.ts